### PR TITLE
Fixed cluster list in burger menu having unnecessary horizontal scroll

### LINKED ISSUE #11809: Burger Menu: Cluster List: Placement of pin cluster is incorrect, causes scroll bar
**Setup**
- Rancher version: dev of 2.10.0

**Describe the bug**
- hovering over a cluster in the burger menu should show an icon on right of row to pin the cluster to the top of the list
- when there are too many clusters a vertical scroll bar is shown
- in vue3 world hovering over a cluster shows the pin icon underneath the vertical scroll bar and prompts a horizontal scroll bar to be shown

**To Reproduce**
- add 4 or so clusters (can be imported/custom)
- reduce height of window such that a scroll bar appears
- hover over the `local` text

**Result**
image

**Expected Result**
- No horizontal scroll bar
- Pin visible
- This is how it previously looked (not great, but better than current)
  - image

**Screenshots**

**Additional context**
- the vertical scroll bar is quite common in my setup
  - legacy enabled, multiple global apps / extensions. chuck in dev tools at bottom and it's always shown

### PR DESCRIPTION
**Pull request by eva-vashkevich**
### Summary
Fixes #11809 

Changed width setting so that it doesn't need scroller on smaller screens. 
Tested on Chrome, Firefox and Safari

### Occurred changes and/or fixed issues

N/A

### Technical notes summary

N/A

### Areas or cases that should be tested

Burger menu:
- Check cluster list with burger menu with 1 cluster both open and collapsed. Try different window sizes.
- Add more clusters (ie import custom clusters) until it requires virtual scroll in the side menu. Repeat step 1.
No horizontal scroll should be visible, clusters should be able to get pinned, layout should seem reasonable.

### Areas which could experience regressions

N/A

### Screenshot/Video
https://github.com/user-attachments/assets/e69cff4c-0597-4750-96dc-2e5884c8fd77
https://github.com/user-attachments/assets/fc64477b-a471-408a-bcc2-1efcc2a42356

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/components/nav/TopLevelMenu.vue [hunk 1]
```diff
@@ -998,7 +998,7 @@ export default {
     }
 
      &.menu-open {
-      width: 300px;
+      width: 315px;
```
Comment (momesgin) [inline shell/components/nav/TopLevelMenu.vue:1001]: Increasing the width will have a side effect on how much TopLevelMenu covers the side bar,

before:
image


after:
image


I think it's fine, but getting an approval from someone in the design team would be nice.

Comment (eva-vashkevich) [reply shell/components/nav/TopLevelMenu.vue:1001]: Good point, I'll check with them

**Comment by eva-vashkevich** [comment]
Updated css. 
Screenshot 2024-11-14 at 1 12 35 PM
Screenshot 2024-11-14 at 1 12 18 PM

**Comment by momesgin** [comment]
> Updated css. Screenshot 2024-11-14 at 1 12 35 PM Screenshot 2024-11-14 at 1 12 18 PM Screenshot 2024-11-14 at 1 16 14 PM

Interestingly I can't see the pins touch the edge of TopLevelMenu on my machine unlike your last screenshot, I tried changing the OS's scroll bar setting but still it doesn't happen. Personally I don't like the pins touching the edge, but at the end of the day that's up to the designers.

Besides that I found two issues after this change, tooltip position and pin icons distance from the edge

before:
image


after:
image

**Review by momesgin — APPROVED** [review]
LGTM

**Pull request merged**